python/benchmark/gen_data_distributed.py

```diff
@@ -0,0 +1,554 @@
+        )
+        self._parser.add_argument(
+            "--no_shutdown",
+            action="store_true",
+            help="do not stop spark session when finished",
+        )
+        self._parser.add_argument(
+            "--test",
+            action="store_true",
+            help="test mode: return blob centers/regression coefficients for test evaluation; only to be used in unit tests",
+        )
+
+        def _restrict_train_size(x: float) -> float:
+            # refer to https://stackoverflow.com/a/12117065/1928940
+            try:
+                x = float(x)
+            except ValueError:
+                raise argparse.ArgumentTypeError(f"{x} is not a floating-point literal")
+
+            if x < 0.0 or x > 1.0:
+                raise argparse.ArgumentTypeError(f"{x} is not in range [0.0, 1.0]")
+
+            return x
+
+        self._parser.add_argument(
+            "--train_fraction",
+            type=_restrict_train_size,  # type: ignore
+            help="the value should be between 0.0 and 1.0 and represent "
+            "the proportion of the dataset to include in the train split",
+        )
+
+        self._add_extra_parameters()
+
+        self.args_: Optional[argparse.Namespace] = None
+
+    def _add_extra_parameters(self) -> None:
+        self.supported_extra_params = self._supported_extra_params()
+        for name, value in self.supported_extra_params.items():
+            if value is None:
+                raise RuntimeError("Must convert None value to the correct type")
+            elif type(value) is type:
+                # value is already type
+                self._parser.add_argument("--" + name, type=value)
+            elif type(value) is bool:
+                self._parser.add_argument("--" + name, type=to_bool)
+            else:
+                # get the type from the value
+                self._parser.add_argument("--" + name, type=type(value))
+
+    def _supported_extra_params(self) -> Dict[str, Any]:
+        """Function to inspect the specific function to get the parameters and values"""
+        return {}
+
+    def _parse_arguments(self, argv: List[Any]) -> None:
+        """Subclass must call this function in __init__"""
+        self.args_ = self._parser.parse_args(argv)
+
+        self.num_rows = self.args_.num_rows
+        self.num_cols = self.args_.num_cols
+        self.dtype = np.dtype(self.args_.dtype)
+
+        self.pyspark_type = dtype_to_pyspark_type(self.dtype)
+        self.feature_cols: List[str] = [f"c{i}" for i in range(self.num_cols)]
+        self.schema = [f"{c} {self.pyspark_type}" for c in self.feature_cols]
+
+        self.extra_params = {
+            k: v
+            for k, v in vars(self.args_).items()
+            if k in self.supported_extra_params and v is not None
+        }
+
+    @property
+    def args(self) -> Optional[argparse.Namespace]:
+        return self.args_
+
+
+class DefaultDataGen(DataGenBase):
```

Comment:
Seems no need to reimplement this class because it can generate datasets distributedly.